sql/parser/eval.go

```diff
@@ -1214,11 +1434,61 @@ func evalCastExpr(expr *CastExpr) (Datum, error) {
 		}
 		return s, nil
 
-		// TODO(pmattis): unimplemented.
-		// case *DecimalType:
-		// case *DateType:
-		// case *TimestampType:
+	// TODO(pmattis): unimplemented.
+	// case *DecimalType:
+	case *DateType:
+		switch d.(type) {
+		case DString:
+			return ParseDate(d.(DString))
+		}
+	case *TimestampType:
+		switch d.(type) {
+		case DString:
+			return ParseTimestamp(d.(DString))
+		}
+	case *IntervalType:
+		switch d.(type) {
+		case DString:
+			d, err := time.ParseDuration(string(d.(DString)))
```

Comment:
I think `time.ParseDuration` only parses a go-specific set of duration notations. Add a `TODO` that we'll likely have to expand the parsing of intervals (e.g. I see postgres supports stuff like `"4 hours 3 minutes"`).